lightwood/helpers/imputers.py

```diff
@@ -0,0 +1,89 @@
+from copy import deepcopy
+from typing import List
+
+import pandas as pd
+
+
+class BaseImputer:
+    def __init__(self, target: str, dependencies: List[str] = []):
+        """
+        Lightwood imputers will fill in missing values in any given column.
+        
+        The single method to implement, `impute`, is where the logic for filling in missing values has to be specified.
+        
+        Note that if access to other columns is required, this can be specified with the `dependencies` parameter.
+
+        :param target: Column that the imputer will modify.
+        :param dependencies: Any additional columns (other than the target) that will be needed inside `impute()`.
+        """  # noqa
+        self.target = target
+        self.dependencies = dependencies
+
+    def impute(self, data: pd.DataFrame) -> pd.DataFrame:
+        raise NotImplementedError("Please implement your custom imputing logic in a module that inherits this class.")
+
+
+class NumericalImputer(BaseImputer):
+    def __init__(self, target: str, dependencies: List[str] = [], value: str = 'zero', typecast: str = None):
+        """
+        Imputer for numerical columns. Supports a handful of different approaches to define the imputation value.
+        
+        String to invoke this class from the cleaner is "numerical.$value", with "value" one of the valid options defined below.
+        
+        :param value: The value to impute. One of 'mean', 'median', 'zero', 'mode'.
+        :param typecast: Used to cast the column into either 'int' or 'float' (`None` skips forced casting).
+        """  # noqa
+        self.value = value
+        self.typecast = typecast
+        super().__init__(target, dependencies)
+
+    def impute(self, data: pd.DataFrame) -> pd.DataFrame:
+        data = deepcopy(data)
```

Comment:
Consider cleaning the data here and potentially importing some primitives from the statistical analysis module for doing so (or make them shared?)